### PR TITLE
Skip "missing" students in submission ZIP

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Skip "missing" students in submission ZIP

JIRA: [AU-74](https://openedx.atlassian.net/browse/AU-74)

**What changed?**

If a student isn't in the student to username map, it means the submission was made referencing a nonexistant User. The only known situation in which this can happen is if a course staff creates a submission from the studio author view.

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

Before fix, first commit:
Create an ora, release it, and make a submission through the studio author view.
Then navigate to the instructor dashboard and generate the submission zip.

Check the lms logs. There should be a report of a keyerror.

After fix, second commit:
Generate the file again. It should generate and show up on the instructor dashboard (although you may not be able to download it dur to devstack-yness

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/data.py [hunk 1]
```diff
@@ -1248,6 +1248,12 @@ def collect_ora2_submission_files(cls, course_id):
         logger.info("[%s] Loaded student identifiers (len=%d)", course_id, len(student_identifiers_map))
 
         for student, submission, _ in all_submission_information:
+            # If course staff creates a submission from the studio preview, they will create a submission for
+            # a student called `student` with no mapping to a real django User
+            if student['student_id'] not in student_identifiers_map:
+                logger.info("[%s] Student id %s has no mapping to a user and will be skipped")
```
Comment (leangseu-edx) [inline openassessment/data.py:1254]: Variables?

Comment (jansenk) [reply openassessment/data.py:1254]: !! good catch

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍
